emulator/src/kreator/assembler.rs

```diff
@@ -0,0 +1,101 @@
+use std::fmt;
+
+struct Assembler {
+    code: Vec<String>,
+    program_counter: u16,
+}
+
+impl Assembler {
+    fn new(input_code: &str) -> Assembler {
+        let mut lines: Vec<String> = Vec::new();
+        for line in input_code.split("\n") {
```

Comment:
This does not work correctly for windows linebreaks, not sure if that matters to us, but it should maybe be fixed.

Reply:
Coincidentally this works on windows linebreaks too (see the new test case). This is the case because trim() removes whitespaces according to the [unicode specification](https://en.wikipedia.org/wiki/Whitespace_character#Unicode) which includies the remaining carriage returns.